src/ensembl/src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewSelectors.ts

```diff
@@ -20,28 +20,28 @@ import {
   getEntityViewerActiveEnsObjectId
 } from 'src/content/app/entity-viewer/state/general/entityViewerGeneralSelectors';
 import {
-  defaultEntityViewerGeneViewUIState,
+  EntityViewerGeneViewUIState,
   transcriptsTabData,
   GeneViewTabMap,
   GeneViewTabData,
   SelectedTabViews
 } from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewState';
 
-export const getGeneViewState = (state: RootState) => {
+export const getGeneViewState = (
+  state: RootState
+): EntityViewerGeneViewUIState | undefined => {
   const activeGenomeId = getEntityViewerActiveGenomeId(state);
   const activeObjectId = getEntityViewerActiveEnsObjectId(state);
 
   if (!activeGenomeId || !activeObjectId) {
-    return defaultEntityViewerGeneViewUIState;
+    return;
   }
-  return (
-    state.entityViewer.geneView?.[activeGenomeId]?.[activeObjectId] ||
-    defaultEntityViewerGeneViewUIState
-  );
+
+  return state.entityViewer.geneView?.[activeGenomeId]?.[activeObjectId];
```

Comment:
1) I am confused about the question mark after geneView (I understand that it was inherited from the previous code). `state.entityViewer.geneView` can never be null or undefined, can it?

2) I am a bit confused, too, by why you decided against returning the default state if geneView[activeGenomeId]?.[activeObjectId] is undefined. Is it better?

Reply:
> I am confused about the question mark after geneView (I understand that it was inherited from the previous code). state.entityViewer.geneView can never be null or undefined, can it?

That's right. It cannot be undefined or null.

> I am a bit confused, too, by why you decided against returning the default state if geneView[activeGenomeId]?.[activeObjectId] is undefined. Is it better?

When we retrieve the value for `viewInRedux` during the initial render in `GeneView.tsx`, we were returning the default value which is actually not from redux. 
As a result, the state does not get updated here:
https://github.com/Ensembl/ensembl-client/pull/315/files#diff-6d9c2d66030239cb4b481357e47aa671R187

Reply:
> That's right. It cannot be undefined or null.

Let's delete the question mark after `state.entityViewer.geneView?` then?

Reply:
Sorry, I thought I had removed it. That's done now.